File: javascript/9996.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");

const [_, pattern] = [input.shift(), input.shift().split("*")];
const regex = new RegExp(String.raw`^${pattern[0]}.*${pattern.at(-1)}$`);

input.forEach((line) => {
  console.log(regex.test(line) ? "DA" : "NE");
});

File: c#/9996.cs
using System;
using System.Text.RegularExpressions;

class Program
{
    static void Main()
    {
        int N = int.Parse(Console.ReadLine());
        string[] Pattern = Console.ReadLine().Split('*');

        Regex regex = new(@$"^{Pattern[0]}.*{Pattern[^1]}+$", RegexOptions.Compiled);

        foreach (var line in Enumerable.Range(0, N).Select(_ => Console.ReadLine()))
        {
            Console.WriteLine(regex.IsMatch(line) ? "DA" : "NE");
        }
    }
}
